migration_challenge_keras_image/Local Notebook.ipynb

```diff
@@ -138,39 +138,67 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import os\n"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "%%time\n",
+    "%time\n",
+    "\n",
     "!rm -rf data/train\n",
     "!rm -rf data/test\n",
+    "import os\n",
+    "import concurrent.futures\n",
+    "from random import randint\n",
+    "\n",
+    "[os.makedirs(os.path.join(\"data/train\", \"digit-%d\" % ix), exist_ok=True) for ix in range(10)]\n",
+    "[os.makedirs(os.path.join(\"data/test\", \"digit-%d\" % ix), exist_ok=True) for ix in range(10)] \n",
+    "\n",
     "\n",
     "# This can take a while due to the number of images\n",
     "def save_to_disk(x, y, base_folder):\n",
     "    \"\"\"Save an image classification dataset to disk as JPEGs in label-named folders\"\"\"\n",
     "    for ix in range(len(y)):\n",
-    "        label_str = \"digit-%d\" % y[ix]\n",
-    "        os.makedirs(os.path.join(base_folder, label_str), exist_ok=True)\n",
+    "        label_dir = \"digit-%d\" % y[ix]\n",
+    "        label_str = \"digit-%d-%d\" % (randint(0, 100000), y[ix])\n",
```

Comment:
I think it would be nicer to try and preserve deterministic outputs if poss - which seems like it'd be easy enough to do by adding an (optional) shard argument to the `save_to_disk` function? Then can pass through the split index as the shard, and save files like `digit-0-0-000001.png`.

Probably has little/no impact on performance I guess because (even though generating random numbers can be slow) we shouldn't be CPU-bound...

Reply:
Restored determinism in 59fc64ff88a367f472b73fe52b3e43622bcf4c26